Lambda-School-Labs/mydish-fe
# Feature/clean up recipe for post

This PR adds functionality to clean up a recipe before it is posted. 

In this context, "clean up" includes the following actions:
- delete any `newline` characters in text input fields
- delete any ingredients, instructions, or notes that are empty (an ingredient is considered empty unless _all three_ ingredient fields (amount, units, name) have a value)
- prevent submission and display error messages if user presses "Save" while any required field is empty.

**To test this branch:**
1. Edit an existing recipe.
2. Delete required data and then try saving. The app should stop you and display error messages.
3. Try saving with `newline` characters in fields for ingredient name, instructions, or notes. The `newlined` should be removed in the saved recipe.
4. Repeat steps **2** and **3** while creating a new recipe. (Because this PR refactors a number of clean-up functions to be usable for both new recipes and existing recipes, it should be tested for both, too.)
